src/base/bittorrent/session.cpp

```diff
@@ -1710,7 +1720,12 @@ void Session::handleTorrentFinished(TorrentHandle *const torrent)
 void Session::handleTorrentResumeDataReady(TorrentHandle *const torrent, const libtorrent::entry &data)
 {
     --m_numResumeData;
-    writeResumeDataFile(torrent, data);
+
+    QByteArray out;
+    libt::bencode(std::back_inserter(out), data);
```

Comment:
I have two main reasons:
1. Because in most cases we save several .fastresume files in a row, if we encode it in IO thread, we will increase the total saving time (total = encode1 + save1 + ... encodeN + saveN). So I prefer to encode it parallel with IO operations.
2. When we invoke method in separate thread we need to copy its arguments. In the case of Qt implicitly shared types (QByteArray) it is quite low-cost operation. And in the case of libtorrent::entry we will have to do a deep copy (thus further increasing the total saving time).